src/app/app.component.ts

```diff
@@ -174,17 +174,34 @@ export class AppComponent implements OnInit, OnDestroy {
     const routesWithoutParent = [];
     extensionRoutes.forEach((extensionRoute: ExtensionRoute) => {
       if (this.extensionRouteHasChild(extensionRoute)) {
-        const routeIndex = this.router.config.findIndex(
-          route => route.path === extensionRoute.parentRoute
-        );
-        this.convertExtensionRouteToRoute(extensionRoute);
-        this.router.config[routeIndex].children.unshift(extensionRoute);
+        // const routeIndex = this.router.config.findIndex(
+        //   route => route.path === extensionRoute.parentRoute
+        // );
+        // this.convertExtensionRouteToRoute(extensionRoute);
+        // this.router.config[routeIndex].children.unshift(extensionRoute);
+
+
+        const parentRoute = this.findRouteRecursively(extensionRoute.parentRoute);
+        if (parentRoute) {
+          this.convertExtensionRouteToRoute(extensionRoute);
+          parentRoute.children.unshift(extensionRoute);
+        }
       } else {
         routesWithoutParent.push(extensionRoute);
       }
     });
 
     this.router.config.unshift(...routesWithoutParent);
+    this.router.resetConfig(this.router.config);
+    console.log(this.router.config);
```

Comment:
note this one should not get merged :)

Reply:
Yepp, it won't, as soon as I finish this PR, I am going to remove the draft mark and ask for a review, please ignore this PR until then :)